samples/java-spring-tracing/src/main/java/com/example/tracing/api/GetRandomNameAction.java

```diff
@@ -0,0 +1,70 @@
+package com.example.tracing.api;
+
+import com.example.tracing.domain.UserEvent;
+import io.opentelemetry.api.trace.*;
+import io.opentelemetry.context.Scope;
+import kalix.javasdk.action.Action;
+import kalix.javasdk.annotations.Subscribe;
+import kalix.javasdk.client.ComponentClient;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.util.concurrent.CompletableFuture;
+
+
+@Subscribe.EventSourcedEntity(value = UserEntity.class, ignoreUnknown = true)
+public class GetRandomNameAction extends Action {
+
+  private static final org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(GetRandomNameAction.class);
+
+  private final ComponentClient componentClient;
+
+  private final WebClient webClient;
+
+  public GetRandomNameAction(ComponentClient componentClient) {
+    this.componentClient = componentClient;
+    this.webClient = WebClient.create("https://randomuser.me/api/?inc=name&noinfo");
+  }
+
+  public Effect<String> handleAdd(UserEvent.UserAdded userAdded) {
+    if (actionContext().eventSubject().isPresent()) {
+      var randomNameFut = getRandomNameAsync();
+
+      var updateNameCall = randomNameFut.thenCompose(name ->
```

Comment:
maybe you could inline the method call, the `randomNameFut` is not used elsewhere